### DISCUSSION TITLE
Argument of type 'AxiosDefaults' is not assignable to parameter of type 'CreateAxiosDefaults<any>'

### DISCUSSION THREAD

**Discussion by rogue-wild** [axios/axios#5737] [General]
` import axios, { AxiosInstance, AxiosResponse } from "axios";

interface Post {
  userId: number;
  id: number;
  title: string;
  body: string;
}

interface AxiosDefaults {
    baseURL: string;
  }
  
  const baseURL: AxiosDefaults = {
    baseURL: "https://jsonplaceholder.typicode.com",
  };
  
  const api: AxiosInstance = axios.create(baseURL);
api.interceptors.response.use(
  (response) => response,
  (error) => {
    console.log(error);
    throw error;
  }
);

export const fetchPost = async (postId: number): Promise<Post> => {
  try {
    const response: AxiosResponse<Post> = await api.get(`/posts/${postId}`);
    const post = response.data;
    return post;
  } catch (error) {
    console.log(error);
    throw error;
  }
};
 `

I'm getting a error in this can anyone help please?

Argument of type 'AxiosDefaults' is not assignable to parameter of type 'CreateAxiosDefaults<any>'.
  Property 'string' is missing in type 'AxiosDefaults' but required in type 'CreateAxiosDefaults<any>'.
   16 |   };
  17 |   
  18 |   const api: AxiosInstance = axios.create(baseURL);
       |                                           ^^^^^^^
  19 | api.interceptors.response.use(
  20 |   (response) => response,
  21 |   (error) => {

**Comment by chinesedfan** [comment]
Change line 18 to,
```
const api: AxiosInstance = axios.create({ baseURL });
```

**Reply by rogue-wild** [reply]
still not working, thank you for the help though

**Reply by chinesedfan** [reply]
Sorry, I misread following lines.  Which version of axios do you use? axios@1.4.0 works well for me.
```
const baseURL: AxiosDefaults = {
    baseURL: "[jsonplaceholder.typicode.com](https://jsonplaceholder.typicode.com/)",
};
```

**Reply by rogue-wild** [reply]
Thank you for your reply, your suggestion worked for me this time.
